stl/inc/typeindex

```diff
@@ -40,9 +41,12 @@ public:
 
 #if _HAS_CXX20
     _NODISCARD strong_ordering operator<=>(const type_index& _Right) const noexcept {
-        return *_Tptr == *_Right._Tptr      ? strong_ordering::equal
-             : _Tptr->before(*_Right._Tptr) ? strong_ordering::less
-                                            : strong_ordering::greater;
+        // TRANSITION, DevCom-10326599, should rely on a stable interface
+        if (_Tptr == _Right._Tptr) {
+            return strong_ordering::equal;
+        }
+
+        return _CSTD strcmp(_Tptr->raw_name() + 1, _Right._Tptr->raw_name() + 1) <=> 0;

```

Comment:
Could we simply `static_assert` that `type_info` and `__std_type_info_data` have the same size and alignment, then reinterpret and call `__std_type_info_compare()`?

Reply:
Their sizes are not equal because of the vptr. And I think `reinterpret_cast` is wrong because the vptr is located at the beginning of `type_info`.

Reply:
Argh, that useless vptr (LWG-2398). Yeah, adding an offset would be too squirrelly. Ok, thanks.